app/src/main/java/com/example/cactusnotes/service/Retrofit.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.cactusnotes.service
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+//QUESTIONS
+//Q1 Why do we use file instead of class ?
+// I dont understand why we use this phrase lazy
+val api: SignupAPI by lazy {
+    Retrofit.Builder()
+        // builder and passing our base url
+        .baseUrl("https://apps.cactus.school")
+        // as we are sending data in json format so we have to add Gson converter factory
+        .addConverterFactory(GsonConverterFactory.create())
+        //we are building our retrofit builder
+        .build()
+        // Create Service
+        .create(SignupAPI::class.java)
```

Comment:
The name of the api shouldn't be signup api since it will be used for all api interaction, like register, login, create, read, update and delete notes.

A better name can be: Api, NotesApi, CactusNotesApi, etc.